src/webgpu/api/validation/gpu_external_texture_expiration.spec.ts

```diff
@@ -0,0 +1,324 @@
+      });
+
+      t.submitCommandBuffer(bindGroup, false);
+
+      // Import again in the same task scope should return same object.
+      const shouldBeTheSameExternalTexture = t.device.importExternalTexture({
+        /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+        source: source as any,
+      });
+      assert(externalTexture === shouldBeTheSameExternalTexture);
+
+      t.submitCommandBuffer(bindGroup, false);
+    });
+  });
+
+g.test('import_and_use_in_different_microtask')
+  .desc(
+    `
+    Tests that in the same task scope, imported GPUExternalTexture is valid in
+    different microtasks.
+    `
+  )
+  .params(u =>
+    u //
+      .combine('sourceType', ['VideoElement', 'VideoFrame'] as const)
+  )
+  .fn(async t => {
+    const sourceType = t.params.sourceType;
+    const videoElement = t.getDefaultVideoElementAndCheck();
+
+    let bindGroup: GPUBindGroup;
+    let externalTexture: GPUExternalTexture;
+    await startPlayingAndWaitForVideo(videoElement, async () => {
+      const source =
+        sourceType === 'VideoFrame'
+          ? await getVideoFrameFromVideoElement(t, videoElement)
+          : videoElement;
+
+      const importMicrotask = Promise.resolve().then(() => {
+        externalTexture = t.device.importExternalTexture({
+          /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+          source: source as any,
+        });
+      });
+
+      const submitMicrotask = Promise.resolve().then(() => {
+        bindGroup = t.device.createBindGroup({
+          layout: t.getDefaultBindGroupLayout(),
+          entries: [{ binding: 0, resource: externalTexture }],
+        });
+        t.submitCommandBuffer(bindGroup, false);
+      });
+      await importMicrotask;
+      await submitMicrotask;
+    });
+  });
+
+g.test('import_and_use_in_different_task')
+  .desc(
+    `
+    Tests that in the different task scope, privious imported GPUExternalTexture
+    should be expired.
+    `
+  )
+  .params(u =>
+    u //
+      .combine('sourceType', ['VideoElement', 'VideoFrame'] as const)
+  )
+  .fn(async t => {
+    const sourceType = t.params.sourceType;
+    const videoElement = t.getDefaultVideoElementAndCheck();
+
+    let bindGroup: GPUBindGroup;
+    let externalTexture: GPUExternalTexture;
+    await startPlayingAndWaitForVideo(videoElement, async () => {
+      const source =
+        sourceType === 'VideoFrame'
+          ? await getVideoFrameFromVideoElement(t, videoElement)
+          : videoElement;
+      externalTexture = t.device.importExternalTexture({
+        /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+        source: source as any,
+      });
+
+      bindGroup = t.device.createBindGroup({
+        layout: t.getDefaultBindGroupLayout(),
+        entries: [{ binding: 0, resource: externalTexture }],
+      });
+
+      t.submitCommandBuffer(bindGroup, false);
+    });
+
+    await waitForNextAnimationFrame(() => {
```

Comment:
The next animation frame could be in a long time. What about doing a setTimeout(0)?